pyOCD/tools/gdb_server.py

```diff
@@ -222,10 +223,15 @@ def run(self, args=None):
         self.args = self.build_parser().parse_args(args)
         self.gdb_server_settings = self.get_gdb_server_settings(self.args)
         self.setup_logging(self.args)
-
         self.process_commands(self.args.commands)
 
         gdb = None
+        if self.args.board_index is not None:
+            ind = int(self.args.board_index)
+            all_mbeds = MbedBoard.getAllConnectedBoards(close=True, blocking=False)
```

Comment:
This list needs to be sorted by unique_id so that the numbering is deterministic (at least if no new boards are attached).  I'm not sure if that makes more sense to do here or in getAllConnectedBoards.